AIMSICD/src/main/java/com/SecUpwN/AIMSICD/utils/Device.java

```diff
@@ -159,29 +147,24 @@ public int getPhoneID() {
         return mPhoneID;
     }
 
+    private String getSimInformation(String simInfo, String errorMsg) {
+        try {
+            if(simInfo != null && !simInfo.isEmpty())
+                return simInfo;
+        } catch (Exception e) {
+            // SIM methods can cause Exceptions on some devices
+            log.error(String.format("%s %s", errorMsg, e.toString()));
+        }
+        return "N/A";
```

Comment:
Just a small note about the translation : for french it shoud be ```N/D``` (Non disponible), but it's not an often used shrinkage in french. It would be better to write ```Indisponible```` (=Unavariable). But it's maybe too long.... i don't know if there's a max length for this text? if yes, keep ```N/D```, it will be ok :p